pkg/verify/store/parallel/parallel.go

```diff
@@ -0,0 +1,92 @@
+// Package parallel combines several signature stores in a single store.
+// Signatures are searched in each substore simultaneously until a
+// match is found.
+package parallel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/cluster-version-operator/pkg/verify/store"
+)
+
+type signatureResponse struct {
+	signature []byte
+	errIn     error
+}
+
+// Store provides access to signatures stored in sub-stores.
+type Store struct {
+	Stores []store.Store
+}
+
+// Signatures fetches signatures for the provided digest.
+func (s *Store) Signatures(ctx context.Context, name string, digest string, fn store.Callback) error {
+	nestedCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	responses := make(chan signatureResponse, len(s.Stores))
+	errorChannelCount := 0
+	errorChannel := make(chan error, 1)
+
+	for i := range s.Stores {
+		errorChannelCount++
+		go func(ctx context.Context, wrappedStore store.Store, name string, digest string, responses chan signatureResponse, errorChannel chan error) {
+			errorChannel <- wrappedStore.Signatures(ctx, name, digest, func(ctx context.Context, signature []byte, errIn error) (done bool, err error) {
+				select {
+				case <-ctx.Done():
+					return true, nil
+				case responses <- signatureResponse{signature: signature, errIn: errIn}:
+				}
+				return false, nil
+			})
+		}(nestedCtx, s.Stores[i], name, digest, responses, errorChannel)
+	}
+
+	allDone := false
+	var loopError error
+	for errorChannelCount > 0 {
+		if allDone {
+			err := <-errorChannel
+			errorChannelCount--
+			if loopError == nil && err != nil && err != context.Canceled && err != context.DeadlineExceeded {
+				loopError = err
+			}
+		} else {
+			select {
+			case response := <-responses:
+				done, err := fn(ctx, response.signature, response.errIn)
+				if done || err != nil {
```

Comment:
The current implementation of fn always returns err nil but looking into the future why would we not want to continue trying to verify even if it did return an error? Hard to say without knowing the nature of the error whther it would be fruitless but I would think all we care about is that one of the signatures satisfies verification.

Reply:
`fn` is a passed-in callback.  Are you taking issue with the parallel handling, or with the wrapped HTTP implementation?  This particular line is "if the caller is satisfied with this most-recent sig, or they considered `errIn` or something else a fatal error, we're done, and can wrap up the parallel handling".

Reply:
Ok, this is me again at the wrong "level", i.e. not at the local Store struct level, and thinking that if someone came along and implemented a different callback fn this if check would short-circuit checking the remaining signatures in the given store, e.g. httpStore, but that's not the case. 

So yeah, we're good here.